src/common/Themes/theme_helpers.cpp

```diff
@@ -0,0 +1,79 @@
+// Port Based on https://stackoverflow.com/a/62811758/5001796
+#include "theme_helpers.h"
+#include "dwmapi.h"
+#include <windows.h>
+#include <iostream>
+#include <vector>
+
+#define STATUS_SUCCESS 0x00000000
+#define DWMWA_USE_IMMERSIVE_DARK_MODE 20
+#define HKEY_WINDOWS_THEME L"Software\\Microsoft\\Windows\\CurrentVersion\\Themes\\Personalize"
+
+typedef NTSTATUS(WINAPI* RtlGetVersionPtr)(PRTL_OSVERSIONINFOW);
+
+namespace
+{
+    RTL_OSVERSIONINFOW GetRealOSVersion()
```

Comment:
Prefer VerifyVersionInfoW and _make sure your app manifest says you run on Windows 10_. There's no need to use internal ntdll APIs.

Reply:
But again, earlier comment about just calling the API and seeing if it fails. No big deal if it does, it won't break anything!

Reply:
Removed all of this logic!